core/src/main/java/org/jdbi/v3/core/LazyHandleSupplier.java

```diff
@@ -13,36 +13,40 @@
  */
 package org.jdbi.v3.core;
 
+import java.util.Deque;
+import java.util.LinkedList;
 import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.extension.ExtensionContext;
 import org.jdbi.v3.core.extension.ExtensionMethod;
 import org.jdbi.v3.core.extension.HandleSupplier;
+import org.jdbi.v3.core.internal.MemoizingSupplier;
 import org.jdbi.v3.core.internal.OnDemandHandleSupplier;
 
-import static org.jdbi.v3.core.internal.Invocations.invokeWith;
-
 class LazyHandleSupplier implements HandleSupplier, AutoCloseable, OnDemandHandleSupplier {
 
     private final Object[] lock = new Object[0];
-
     private final Jdbi db;
-    private final ThreadLocal<ConfigRegistry> localConfig;
-
-    @SuppressWarnings("ThreadLocalUsage")
-    private final ThreadLocal<ExtensionMethod> localExtensionMethod = new ThreadLocal<>();
-
-    private volatile Handle handle;
-    private volatile boolean closed = false;
+    private final AtomicBoolean closed = new AtomicBoolean();
```

Comment:
We don't seem to actually use any of the added features `AtomicBoolean` brings over the `volatile` we had before.